### PR TITLE
Groovy - special handling of double-quote values in enum arguments

### LINKED ISSUE #6015: Groovy parsing infinite loop with URLs used in Enum arguments
## What version of OpenRewrite are you using?

Current main = 9f1f7c3301288cdcb7c297d4762077fdfde1794c

## What is the smallest, simplest way to reproduce the problem?
```java
    @Test
    void uris() {
        rewriteRun(
          groovy(
            """
            enum E {
              LOCAL("http://localhost:8080/api/v1/clusters")

              E(String uri) {
              }
            }
            """
          )
        );
    }
```
in `org.openrewrite.groovy.tree.EnumTest`

never stops. I.e. goes into an infinite loop.

## Context

Similar to:
- #6009 
- #6014 

I suspect this is because `sourceBefore(")` is used and on observing `//` it considers the rest of the line as line-comment, which is not appropriate.

## OSS repro

- https://github.com/hortonworks/ambari-rest-client/blob/d6dd0f25ac1e528a72346a7f5479dc509986de55/src/test/groovy/com/sequenceiq/ambari/client/TestResources.groovy#L21

### PR DESCRIPTION
**Pull request by greg-at-moderne**
## What's changed?

In Groovy handling double-quoted arguments (i.e. Strings) in enum arguments in a different way than the other arguments. Here we shouldn't use the `sourceBefore()` hack which is designed to handle whitespace. Manual `cursor` advancement instead.

## What's your motivation?

- fixes #6015

### CODE REVIEW

File: rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java [hunk 1]
```diff
@@ -467,7 +467,12 @@ private J.EnumValue visitEnumField(FieldNode field) {
                     Delimiter delimiter = getDelimiter(null, cursor);
                     if (delimiter != null) {
                         cursor += delimiter.open.length();
-                        sourceBefore(delimiter.close);
+                        if (delimiter.close == "\"") {
```
Comment (timtebeek) [inline rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java]: Did you mean the following? Or was this intentional? Rarely a problem in practice I think, but it looked out of place.
```suggestion
                        if ("\"".equals(delimiter.close)) {
```

File: rewrite-groovy/src/test/java/org/openrewrite/groovy/tree/EnumTest.java [hunk 2]
```diff
@@ -277,4 +277,22 @@ private TagNameSerializer(String type, Function<String, String> fun) {
...
     }
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/6015")
+    @Test
+    void uris() {
+        rewriteRun(
+          groovy(
+            """
+            enum E {
+              LOCAL("http://localhost:8080/api/v1/clusters")
```
Comment (timtebeek) [inline rewrite-groovy/src/test/java/org/openrewrite/groovy/tree/EnumTest.java]: Looks like we might have another edge case with
```
              ESCAPED_QUOTE("http://localhost:8080/\"api/v1/clusters")
```

Comment (greg-at-moderne) [reply rewrite-groovy/src/test/java/org/openrewrite/groovy/tree/EnumTest.java]: Good point.
aff2617e7

**Review by timtebeek — APPROVED** [review]
Thanks for the fixes here!
